src/Amadevus.RecordGenerator.Generators/EqualityGenerators.cs

```diff
@@ -133,21 +133,26 @@ private static BlockSyntax GenerateStandaloneObjectEquals(RecordDescriptor descr
 
         private static BlockSyntax GenerateForwardToEquatableEquals(RecordDescriptor descriptor)
         {
-            // return this.Equals(obj as MyRecord);
+            // return obj is MyRecord castObj && this.Equals(castObj);
+            const string otherVariableName = "other";
             return
                 Block(
                     ReturnStatement(
+                        BinaryExpression(
+                            LogicalAndExpression,
+                            IsPatternExpression(
+                                IdentifierName(objVariableName),
+                                DeclarationPattern(
+                                    descriptor.TypeSyntax,
+                                    SingleVariableDesignation(Identifier(otherVariableName)))),
                         InvocationExpression(
```

Comment:
The indentation here is off. `InvocationExpression` should align with `IsPatternExpression` as an argument of `BinaryExpression`.